app/gui/view/src/debug_mode_popup.rs

```diff
@@ -0,0 +1,208 @@
+//! Editor.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::animation::delayed::DelayedAnimation;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::Animation;
+use ensogl_component::label::Label;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Mitigate limitations of constant strings concatenation.
+macro_rules! define_debug_mode_shortcut {
+    ($shortcut:literal) => {
+        /// A keyboard shortcut used to enable/disable Debug Mode.
+        pub const DEBUG_MODE_SHORTCUT: &str = $shortcut;
+        const DEBUG_MODE_ENABLED: &str =
+            concat!("Debug Mode enabled. To disable, press `", $shortcut, "`.");
+    };
+}
+define_debug_mode_shortcut!("ctrl shift d");
+const DEBUG_MODE_DISABLED: &str = "Debug Mode disabled.";
+
+const LABEL_VISIBILITY_DELAY_MS: f32 = 3_000.0;
+const LABEL_PADDING_TOP: f32 = 50.0;
+
+
+
+// ==================
+// === PopupLabel ===
+// ==================
+
+/// Text label that disappears after a predefined delay.
+#[derive(Debug, Clone, CloneRef)]
+struct PopupLabel {
+    label:   Label,
+    network: frp::Network,
+    show:    frp::Source,
+    hide:    frp::Source,
+}
+
+impl display::Object for PopupLabel {
+    fn display_object(&self) -> &display::object::Instance<Scene> {
+        self.label.display_object()
+    }
+}
+
+impl PopupLabel {
+    /// Constructor. Label will have `text` content and will disappear after `delay` milliseconds.
+    pub fn new(app: &Application, text: &str, delay: f32) -> Self {
+        let network = frp::Network::new("PopupLabel");
+        let label = Label::new(app);
+        label.set_content(text);
+        label.set_opacity(0.0);
+        let background_layer = &app.display.scene().layers.panel;
+        let text_layer = &app.display.scene().layers.panel_text;
+        label.set_layers(background_layer, text_layer);
+
+        let opacity_animation = Animation::new(&network);
+        network.store(&opacity_animation);
+        let delay_animation = DelayedAnimation::new(&network);
+        delay_animation.set_delay(delay);
+        delay_animation.set_duration(0.0);
+        network.store(&delay_animation);
+
+        frp::extend! { network
+            show <- source_();
+            hide <- source_();
+
+            delay_animation.start <+ show;
+            delay_animation.reset <+ hide;
+            should_hide <- any(hide, delay_animation.on_end);
+
+            opacity_animation.target <+ show.constant(1.0);
+            opacity_animation.target <+ should_hide.constant(0.0);
+            label.set_opacity <+ opacity_animation.value;
+        }
+
+        Self { label, network, show, hide }
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+#[derive(Debug, Clone, CloneRef)]
+struct Model {
+    display_object: display::object::Instance,
+    enabled_label:  PopupLabel,
+    disabled_label: PopupLabel,
```

Comment:
What about keeping one PopupLabel and resetting text in it?

Reply:
I'll try. I was thinking that it will complicate the implementation a bit because of the additional handling of delay reset, but maybe it's easier than I thought.

Reply:
Done